### PR TITLE
Deploy storybook github action

### PR DESCRIPTION
**Pull request by srallen**
_Please request review from `@zooniverse/frontend` team. If PR is related to design, please request review from `@beckyrother` in addition._ 

Package: N/A

Describe your changes:
This adds an action to deploy the storybooks on push and merge to master. It's unclear to me if I need to add this flag: https://github.com/storybookjs/storybook-deployer#deploying-storybook-as-part-of-a-ci-service I'm guessing no for this, but I need to actually try running this to see what happens.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: .github/workflows/storybook-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    steps:
+      - name: Deploy storybooks
+        with:
+          node-version: ${{ matrix.node-version }}
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: zooniverse/depoy-storybooks-action@v1.0.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: npm install -g yarn
+      - run: yarn deploy-storybook
```
Comment (srallen) [inline .github/workflows/storybook-deploy.yml:37]: I might need to `yarn bootstrap` before this.

File: .github/workflows/storybook-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - name: Deploy storybooks
+        with:
+          node-version: ${{ matrix.node-version }}
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: zooniverse/depoy-storybooks-action@v1.0.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: npm install -g yarn
```
Comment (srallen) [inline .github/workflows/storybook-deploy.yml:36]: Is there a better way to get yarn?

Comment (eatyourgreens) [reply .github/workflows/storybook-deploy.yml:36]: It should be installed with the monorepo, in `.yarn`. The CI tests action has an example of running yarn with node.

**Review by eatyourgreens — COMMENTED** [review]
This looks good. I think Node 14 will fail the engine check in `package.json` and you probably don't need to bootstrap the whole monorepo in order to build the storybooks.

File: .github/workflows/storybook-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - name: Use Node.js 14
+        uses: actions/setup-node@v1
+        with:
+          node-version: '14.x'
```
[the remark this answers is not in the document]
Comment (srallen) [reply .github/workflows/storybook-deploy.yml:26]: That was my plan.

**Comment by srallen** [comment]
This was in draft status. Could you hold off until I ask for review in the future?

**Comment by eatyourgreens** [comment]
Sure!

**Comment by srallen** [comment]
Closing this for now. There are too many PRs open.

**Pull request closed without merging**